### ISSUE TITLE
Adafruit color sensor conversion is incorrect

### ISSUE THREAD

**Issue by calebsander** [ftctechnh/ftc_app#118]
The TCS34725 sensor support in the SDK treats each of the four color values as a signed short, when they are really unsigned shorts. This can easily be tested by moving a sensor very close to a white piece of paper. The value will suddenly skip from around `30000` to `-30000` and will max out at `-1`. This violates the assumptions that light values should be positive and that brighter light should result in higher readings. You can test the conversion with the following Java code:
````java
import java.nio.ByteBuffer;
public class Test {
        public static void main(String[] args) {
                final byte high = (byte)0xFF;
                final byte low = (byte)0xFF;
                System.out.println("FTC Conversion: " + (high << 8 | low & 0xFF));
                System.out.println("Should be: " + (ByteBuffer.wrap(new byte[]{high, low}).getShort() & 0xFFFF));
                System.out.println("Corrected FTC: " + ((high & 0xFF) << 8 | low & 0xFF));
        }
}
````

**Comment by calebsander** [comment]
Still an issue in yesterday's release

**Comment by calebsander** [comment]
Seems fixed in 2.0.
